pyproject.toml

```diff
@@ -16,6 +16,10 @@ pyarrow = "^9.0.0"
 lz4 = "^4.0.2"
 requests=">2.18.1"
 oauthlib=">=3.1.0"
+numpy = [
+    {version = "1.21.1", python = ">=3.7,<3.8"},
+    {version = "1.23.4", python = ">=3.8"}
+]
```

Comment:
why is numpy changes are relevant to staging?

Reply:
This one is a bit tricky to explain. tl;dr in this PR we're making explicit a Python-version-specific numpy dependency that was _implicit_ when we added Python 3.,10 support in #31.

Implicit here means that we didn't specify the numpy versions in `pyproject.toml` but Poetry automatically figured it out during dependency resolution and captured that in `poetry.lock`. Since then, poetry dependency resolution changed. In the process of building this PR, Poetry found itself in a dependency resolution recursive loop. To resolve the issue I explicitly pinned the `numpy` version so that `poetry install` would run properly on Python 3.10.

We _could have done this during #31_ but it wasn't necessary because Poetry worked at that time and all our unit and e2e tests passed.

AFAIK this change doesn't actually affect how the package is installed. It's here to help Poetry know how to resolve its dependencies.

Reply:
Just adding a note here to say that this is an _excellent_ question to ask for this PR. If you peek at the `main` branch you'll see we've been adding more explicit Python-version-specific dependency declarations because of `pyarrow` too.